src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,46 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.*;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        BulldozerProducer bulldozerProducer = new BulldozerProducer();
+        TruckProducer truckProducer = new TruckProducer();
+        ExcavatorProducer excavatorProducer = new ExcavatorProducer();
+
+        List<? extends Machine> machines;
+
+        if (type == Bulldozer.class) {
+            machines = bulldozerProducer.get();
+        } else if (type == Truck.class) {
+            machines = truckProducer.get();
+        } else if (type == Excavator.class) {
+            machines = excavatorProducer.get();
+        } else {
+            return List.of();
+        }
```

Comment:
This if-else construction could be simplified. You don't need `else` after `return` in if clause. Also, instead of creating a variable `machines` and then casting it to `List<Machine>`, you could return the result directly in each if clause. For example:
java
if (type == Bulldozer.class) {
    return new ArrayList<>(bulldozerProducer.get());
}
if (type == Truck.class) {
    return new ArrayList<>(truckProducer.get());
}
if (type == Excavator.class) {
    return new ArrayList<>(excavatorProducer.get());
}
return List.of();
```